Jenkinsfile

```diff
@@ -24,83 +30,436 @@ pipeline {
+              options { skipDefaultCheckout() }
+              steps {
+                makeTarget("Auditbeat oss Mac OS X", "TEST_ENVIRONMENT=0 -C auditbeat testsuite")
+              }
+            }
+
+          }
+        }
+        stage('Libbeat'){
+          agent { label 'ubuntu && immutable' }
+          options { skipDefaultCheckout() }
+          when {
+            anyOf {
+              changeset(glob: 'libbeat/**')
+              expression { return params.runAllStages }
+            }
+          }
+          stages {
+            stage('Libbeat oss'){
+              steps {
+                withBeatsEnv(){
+                  makeTarget("Libbeat oss Linux", "-C libbeat testsuite")
+                }
+              }
+            }
+            stage('Libbeat crosscompile'){
+              steps {
+                withBeatsEnv(){
+                  makeTarget("Libbeat oss crosscompile", "-C libbeat crosscompile")
+                }
+              }
+            }
+            stage('Libbeat stress-tests'){
+              steps {
+                withBeatsEnv(){
+                  makeTarget("Libbeat stress-tests", "STRESS_TEST_OPTIONS='-timeout=20m -race -v -parallel 1' -C libbeat stress-tests")
+                }
+              }
+            }
+            stage('Libbeat x-pack'){
+              steps {
+                withBeatsEnv(){
+                  makeTarget("Libbeat x-pack Linux", "-C x-pack/libbeat testsuite")
+                }
               }
             }
           }
-          post {
-            always {
-              junit(allowEmptyResults: true,
-                keepLongStdio: true,
-                testResults: "${BASE_DIR}/build/junit-*.xml")
+        }
+        stage('Metricbeat'){
+          agent { label 'ubuntu && immutable' }
+          options { skipDefaultCheckout() }
+          when {
+            anyOf {
+              changeset(glob: 'libbeat/**')
+              changeset(glob: 'metricbeat/**')
+              expression { return params.runAllStages }
+            }
+          }
+          stages {
+            stage('Metricbeat Unit tests'){
+              steps {
+                withBeatsEnv(){
+                  makeTarget("Metricbeat Unit tests", "-C metricbeat unit-tests coverage-report")
+                }
+              }
+            }
+            stage('Metricbeat Integration tests'){
+              steps {
+                withBeatsEnv(){
+                  makeTarget("Metricbeat Integration tests", "-C metricbeat integration-tests-environment coverage-report")
+                }
+              }
+            }
+            stage('Metricbeat System tests'){
+              agent { label 'ubuntu && immutable' }
+              options { skipDefaultCheckout() }
+              steps {
+                withBeatsEnv(){
+                  makeTarget("Metricbeat System tests", "-C metricbeat update system-tests-environment coverage-report")
+                }
+              }
+            }
```

Comment:
Could these stages be run in parallel?

Reply:
sure, let's make it works first

Reply:
done, I moved them to a parallel stage, it takes about 1h to build now

Reply:
This is already faster than our current Jenkins builds :slightly_smiling_face: